cryptographic_estimators/MREstimator/MRAlgorithms/support_minors.py

```diff
@@ -0,0 +1,261 @@
+
+    @optimal_parameter
+    def b(self):
+        """
+        Return the optimal `b`, i.e. the degree of the linear variables in the Macaulay matrix
+
+        EXAMPLES::
+
+            sage: from cryptographic_estimators.MREstimator.MRAlgorithms.support_minors import SupportMinors
+            sage: from cryptographic_estimators.MREstimator.mr_problem import MRProblem
+            sage: SM = SupportMinors(MRProblem(q=7, m=9, n=10, k=15, r=4))
+            sage: SM.b()
+            1
+        """
+        return self._get_optimal_parameter(MR_LINEAR_VARIABLES_DEGREE)
+
+    @optimal_parameter
+    def nprime(self):
+        """
+        Return the optimal `nprime`, i.e. the number of columns to be selected
+
+        EXAMPLES::
+
+            sage: from cryptographic_estimators.MREstimator.MRAlgorithms.support_minors import SupportMinors
+            sage: from cryptographic_estimators.MREstimator.mr_problem import MRProblem
+            sage: SM = SupportMinors(MRProblem(q=7, m=9, n=10, k=15, r=4))
+            sage: SM.nprime()
+            4
+        """
+        return self._get_optimal_parameter(MR_REDUCED_NUMBER_OF_COLUMNS)
+
+    @optimal_parameter
+    def variant(self):
+        """
+        Return the optimal `variant`
+
+        EXAMPLES::
+
+            sage: from cryptographic_estimators.MREstimator.MRAlgorithms.support_minors import SupportMinors
+            sage: from cryptographic_estimators.MREstimator.mr_problem import MRProblem
+            sage: SM = SupportMinors(MRProblem(q=7, m=9, n=10, k=15, r=4))
+            sage: SM.nprime()
+            'strassen'
+        """
+        return self._get_optimal_parameter(MR_VARIANT)
+
+
+    def _expected_dimension_of_support_minors_equations(self,q, m, n, K, r, b):
+        """
+        Return the expected number of linearly independent support minors equations
+
+        """
+        if q == 2:
+            temp = 0
+            for j in range(1, b + 1):
+                temp += sum([(-1) ** (i + 1) * binomial(n, r + i) * binomial(m + i - 1, i) * binomial(K, j - i) for i in
+                             range(1, j + 1)])
+        else:
+            temp = sum(
+                [(-1) ** (i + 1) * binomial(n, r + i) * binomial(m + i - 1, i) * binomial(K + b - i - 1, b - i) for i in
+                 range(1, b + 1)])
+        return temp
+
+
+    def _dimension(self,q, n, K, r, b):
+        """
+        Dimension of the smallest vector space spanned by monomials containing the support minors equations
+        """
+        if q == 2:
+            temp = binomial(n, r) * sum([binomial(K, j) for j in range(1, b + 1)])
+        else:
+            temp = binomial(n, r) * binomial(K + b - 1, b)
+        return temp
+
+
+    def _are_parameters_invalid(self, parameters: dict):
+        """
+        Specifies constraints on the parameters
+        """
+        a = parameters["a"]
+        lv = parameters["lv"]
+        b = parameters["b"]
```

Comment:
In case we need multiple values of a dictionary like here, I think we should use

par = SimpleNamespace(**parameters)
to access the variables like par.a, par.lv etc.

(it requires "from types import SimpleNamespace")